usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java

```diff
@@ -24,100 +24,160 @@
-                .build());
+        try {
+            BINARY_EVENT_LOGGER.info("Binary format logging for the following usage event: {}", details);
+            checkNotNull(details);
+            log(eventBuilder()
+                    .setCdpEnvironmentStatusChanged(details)
+                    .build());
+        } catch (Exception e) {
+            BINARY_EVENT_LOGGER.warn("Could not log binary format for the following usage event: {}! Cause: {}", details, e.getMessage());
+        }
     }
 
     @Override
     public void cdpDatalakeRequested(UsageProto.CDPDatalakeRequested details) {
-        checkNotNull(details);
-        log(eventBuilder()
-                .setCdpDatalakeRequested(details)
-                .build());
+        try {
+            BINARY_EVENT_LOGGER.info("Binary format logging for the following usage event: {}", details);
+            checkNotNull(details);
+            log(eventBuilder()
+                    .setCdpDatalakeRequested(details)
+                    .build());
+        } catch (Exception e) {
+            BINARY_EVENT_LOGGER.warn("Could not log binary format for the following usage event: {}! Cause: {}", details, e.getMessage());
+        }
     }
 
     @Override
     public void cdpDatalakeStatusChanged(UsageProto.CDPDatalakeStatusChanged details) {
-        checkNotNull(details);
-        log(eventBuilder()
-                .setCdpDatalakeStatusChanged(details)
-                .build());
+        try {
+            BINARY_EVENT_LOGGER.info("Binary format logging for the following usage event: {}", details);
+            checkNotNull(details);
+            log(eventBuilder()
+                    .setCdpDatalakeStatusChanged(details)
+                    .build());
+        } catch (Exception e) {
+            BINARY_EVENT_LOGGER.warn("Could not log binary format for the following usage event: {}! Cause: {}", details, e.getMessage());
+        }
     }
 
     @Override
     public void cdpDatahubRequested(UsageProto.CDPDatahubRequested details) {
-        checkNotNull(details);
-        log(eventBuilder()
-                .setCdpDatahubRequested(details)
-                .build());
+        try {
+            BINARY_EVENT_LOGGER.info("Binary format logging for the following usage event: {}", details);
+            checkNotNull(details);
+            log(eventBuilder()
+                    .setCdpDatahubRequested(details)
+                    .build());
+        } catch (Exception e) {
+            BINARY_EVENT_LOGGER.warn("Could not log binary format for the following usage event: {}! Cause: {}", details, e.getMessage());
+        }
     }
 
     @Override
     public void cdpDatahubStatusChanged(UsageProto.CDPDatahubStatusChanged details) {
-        checkNotNull(details);
-        log(eventBuilder()
-                .setCdpDatahubStatusChanged(details)
-                .build());
+        try {
+            BINARY_EVENT_LOGGER.info("Binary format logging for the following usage event: {}", details);
```

Comment:
wouldn't it be better to move this into the log method?

Reply:
done